compiler/codegen/cg-expr.cpp

```diff
@@ -5141,8 +5142,7 @@ DEFINE_PRIM(SET_MEMBER) {
 DEFINE_PRIM(CHECK_NIL) {
     GenRet ptr = call->get(1);
 
-    if (ptr.chplType->symbol->hasFlag(FLAG_WIDE_CLASS))
-      ptr = codegenRaddr(ptr);
+    ptr = maybeConvertToLocalPointer(call->get(1), ptr);
```

Comment:
This fixes a bug I ran into where we were incorrectly generating code for PRIM_CHECK_NIL with a wide-reference to a wide-class. This bug was pre-existing, bug only encountered because apparently we don't use ref-intent on class methods very much, and we do specify ``ref`` intent for ``deserialize`` methods.